### ISSUE TITLE
all: create a special annotation to indicate that a function panics

### ISSUE THREAD

**Issue by pjebs** [golang/go#44056]
It would be good for a special comment tag or something to document that a function potentially panics. 

Godoc would detect the tag and display a special label or ⚠️ symbol .

or 
<pre>
<h1>🅿 🅟</h1>

</pre>

**Comment by davecheney** [comment]
What about interfaces?

**Comment by pjebs** [comment]
The special comment tag can apply to functions in an interface too.

However, any exported object that implements a function from the interface can also document panics.

**Comment by ianlancetaylor** [comment]
Many functions can panic if invalid input is passed.

**Comment by pjebs** [comment]
They should all be labeled as potentially panicing.

The rationale is that Go encourages you to do things concurrently in separate goroutines.
Webservers, even with Recovery middleware, can't protect against panics in separate goroutines.

This is very dangerous because it's easy to forget that a function may panic. Labelling it as such is the least we can do, before it brings down the entire server.

**Comment by fzipp** [comment]
One nice property of Go's doc comments is that they are not schematised, and a nice property of godoc & Co. is that the presentation is clutter free. A panic should be simply mentioned in the text as a sentence, just like anything else noteworthy should be mentioned in the text.

**Comment by seankhliao** [comment]
I don't think this is a good idea if it's done voluntarily and manually as this wouldn't cover the myriad of ways things can panic and would lead users into a false sense of security when they don't see the marker.

**Comment by pjebs** [comment]
@seankhliao Perhaps it can be automated? No need for any human intervention.
